### PR TITLE
Fix segmentation violation from MCParticles with duplicate track IDs

### PR DESCRIPTION
**Pull request by knoepfel**
This PR resolves [Redmine issue #26197](https://cdcvs.fnal.gov/redmine/issues/26197), where a segmentation violation is reported due to attempting to access a trajectory point in an `MCParticle` that had none.  The basic problem is that the `larg4` services were not implemented to support reading multiple `MCTruth` collections where one of those collections has multiple `MCTruth` objects in it.  This PR enables that behavior.

I have added some design considerations so that the behavior is less susceptible to these types of errors.  That will be a separate project, however.

### CODE REVIEW

**Comment by knoepfel** [comment]
trigger build

File: larg4/pluginActions/MCTruthEventAction_service.cc [hunk 1]
```diff
@@ -129,27 +111,27 @@ void larg4::MCTruthEventActionService::generatePrimaries(G4Event * anEvent) {
...
+        G4ParticleDefinition* particleDefinition = nullptr;
 
-        if(pdgCode==0){
+        if (pdgCode == 0) {
           particleDefinition = fParticleTable->FindParticle("opticalphoton");
         }
         else
           particleDefinition = fParticleTable->FindParticle(pdgCode);
 
         if ( pdgCode > 1000000000) { // If the particle is a nucleus
```
Comment (nusense) [inline larg4/pluginActions/MCTruthEventAction_service.cc]: this really should be 
 if ( pdgCode > 1000000000 &&  pdgCode < 2000000000) 
 and a separate handling of PDG >=  2000000000 dealt with (i.e. notification that they can't be handled as they're generator specific and not ions)

Comment (nusense) [reply larg4/pluginActions/MCTruthEventAction_service.cc]: "this" being line 145/127

Comment (knoepfel) [reply larg4/pluginActions/MCTruthEventAction_service.cc]: @nusense, thanks for point that out.  @lgarren, although Robert's comment is not directly related to the changes I've made here, do you think there would be any problem with me adjusting the code along the lines he suggests?  I'm guessing the code wouldn't be working correctly anyway if a pdgCode of 2000000000 or greater were encountered...?

Comment (lgarren) [reply larg4/pluginActions/MCTruthEventAction_service.cc]: As long as the change doesn't break anything, I see no problem.

**Comment by knoepfel** [comment]
trigger build

**Comment by lgarren** [comment]
@knoepfel @nusense   It appears that the requested change has resulted in some differences in product sizes.

**Comment by henrylay97** [comment]
Can comment on behalf of SBND that the change seen there is part of an intermittent bug we see in one specific regression test and is not due to this PR.

**Comment by yangtj207** [comment]
+1

**Comment by lgarren** [comment]
approve

**Pull request merged**